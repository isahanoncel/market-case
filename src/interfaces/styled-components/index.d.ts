// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
