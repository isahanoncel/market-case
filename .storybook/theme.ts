import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
