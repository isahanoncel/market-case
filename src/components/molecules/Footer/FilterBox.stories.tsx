import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Footer from '.';

export default {
  title: 'Molecules/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>;

export const Default: ComponentStory<typeof Footer> = (args) => (
  <Footer {...args} />
);
