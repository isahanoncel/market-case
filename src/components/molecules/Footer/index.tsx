import React, { FC } from 'react';
import Flex from '../../atoms/Flex';
import Label from '../../atoms/Label';

const FilterBox: FC = () => {
  return (
    <Flex justifyContent="center" alignItems="center" margin="0 0 87px 0">
      <Label color="primary" size={13} weight={300}>
        ©2019 Market • Privacy Policy
      </Label>
    </Flex>
  );
};

export default FilterBox;
