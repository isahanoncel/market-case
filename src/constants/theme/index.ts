import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
