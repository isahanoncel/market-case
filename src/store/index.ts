import { configureStore } from '@reduxjs/toolkit';
import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux';
import products from './slices/products';
import basket from './slices/basket';

const store = configureStore({
  reducer: {
    products,
    basket,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export default store;
