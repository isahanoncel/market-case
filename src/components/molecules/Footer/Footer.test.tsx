import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Footer from '.';
import theme from '../../../constants/theme';
import store from '../../../store';

test('Footer snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Footer />
        </ThemeProvider>
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
