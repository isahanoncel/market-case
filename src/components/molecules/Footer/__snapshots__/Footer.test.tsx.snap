// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot testing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 87px 0;
}

.c1 {
  font-size: 13px;
  line-height: 14px;
  color: #1EA4CE;
  font-weight: 300;
  margin: 0;
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    height: auto;
  }
}

<div
  className="c0"
>
  <span
    className="c1"
    color="primary"
    size={13}
  >
    ©2019 Market • Privacy Policy
  </span>
</div>
`;
